llvm_passes/HipPromoteInts.cpp

```diff
@@ -0,0 +1,377 @@
+#include "HipPromoteInts.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/raw_ostream.h"
+
+#define DEBUG_TYPE "hip-promote-ints"
+
+/**
+ * This pass promotes integer types to the next standard bit width.
+ * During optimization of loops, LLVM generates non-standard integer types
+ * such as i33 or i56
+ *
+ * Algorithm Overview:
+ * ------------------
+ * The pass uses a two-phase approach to handle non-standard integer types:
+ *
+ * 1. Construction Phase:
+ *    - When encountering a non-standard integer type (e.g., i33), the pass
+first creates
```

Comment:
I guess, clang-format messed up the formatting here.
```suggestion
 *    - When encountering a non-standard integer type (e.g., i33), the pass
        first creates
```